backend/dataset/views.py

```diff
@@ -366,6 +363,116 @@ def projects(self, request, pk):
 
         return Response(serializer.data)
 
+    @action(methods=["GET"], detail=True, name="Get all past instances of celery tasks")
+    def get_async_task_results(self, request, pk):
+        """
+        View to get all past instances of celery tasks
+        URL: /data/instances/<instance-id>/get_async_task_results?task_name=<task-name>
+        Accepted methods: GET
+
+        Returns:
+            A list of all past instances of celery tasks for a specific task
+        """
+
+        # Get the task name from the request
+        task_name = request.query_params.get("task_name")
+
+        # Check if task name is in allowed task names list
+        if task_name not in ALLOWED_CELERY_TASKS:
+            return Response(
+                {
+                    "message": "Invalid task name for this app.",
+                    "allowed_tasks": ALLOWED_CELERY_TASKS,
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Check if the task name has the word projects in it
+        if "projects" in task_name:
+
+            # Get the IDs of the projects associated with the dataset instance
+            project_ids = (
+                apps.get_model("projects", "Project")
+                .objects.filter(dataset_id=pk)
+                .values_list("id", flat=True)
+            )
+
+            # Create the project keywords list
+            project_id_keyword_args = [
+                "'project_id': " + "'" + str(pk) + "'" for pk in project_ids
+            ]
+
+            # Turn list of project ID keywords into list of Q objects
+            queries = [
+                Q(task_kwargs__contains=project_keyword)
+                for project_keyword in project_id_keyword_args
+            ]
+            print("The QUERY", queries)
+            query = queries.pop()
```

Comment:
Please handle with exception when the list is empty

Reply:
I have updated this.